src/compile/scale/range.ts

```diff
@@ -105,7 +105,27 @@ export function parseRangeForChannel(channel: ScaleChannel, model: UnitModel): E
       } else {
         switch (property) {
           case 'range':
-            return makeExplicit(specifiedScale[property]);
+            if (isArray(specifiedScale.range) && (channel === 'x' || channel === 'y')) {
+              return makeExplicit(
+                specifiedScale.range.map(v => {
+                  if (v === 'width' || v === 'height') {
+                    // get signal for width/height
+
+                    // Note that, just like default range logic below, these range signals are temporary
+                    // as they can be merged and renamed.
+                    // (We do not have the right size signal here since parseLayoutSize() happens after parseScale().)
+                    // We will later replace these temporary names with
```

Comment:
I'm confused by this. If we later replace these temporary names, why do we need them? The idea with the SignalRefWrapper is that you don't have to explicitly rename later but that the closure handles the renaming.

Reply:
It’s the same comment from below that i copy from.

Reply:
Ahh, I didn't see this on Github. The comment is wrong then. Please delete it or fix it.